apps/docs/content/guides/auth/auth-mfa/phone.mdx

```diff
@@ -0,0 +1,313 @@
+  const [challengeId, setChallengeId] = useState('')
+  const [phoneNumber, setPhoneNumber] = useState('')
+
+  const startChallenge = async () => {
+    setError('')
+    try {
+      const factors = await supabase.auth.mfa.listFactors()
+      if (factors.error) {
+        throw factors.error
+      }
+
+      const phoneFactor = factors.data.phone[0]
+
+      if (!phoneFactor) {
+        throw new Error('No phone factors found!')
+      }
+
+      const factorId = phoneFactor.id
+      setFactorId(factorId)
+      setPhoneNumber(phoneFactor.phone)
+
+      const challenge = await supabase.auth.mfa.challenge({ factorId })
+      if (challenge.error) {
+        setError(challenge.error.message)
+        throw challenge.error
+      }
+
+      setChallengeId(challenge.data.id)
+    } catch (error) {
+      setError(error.message)
+    }
+  }
+
+  const verifyCode = async () => {
+    setError('')
+    try {
+      const verify = await supabase.auth.mfa.verify({
+        factorId,
+        challengeId,
+        code: verifyCode,
+      })
+      if (verify.error) {
+        setError(verify.error.message)
+        throw verify.error
+      }
+    } catch (error) {
+      setError(error.message)
+    }
+  }
+
+  return (
+    <>
+      <div>Please enter the code sent to your phone.</div>
+      {phoneNumber && <div>Phone number: {phoneNumber}</div>}
+      {error && <div className="error">{error}</div>}
+      <input
+        type="text"
+        value={verifyCode}
+        onChange={(e) => setVerifyCode(e.target.value.trim())}
+      />
+      {!challengeId ? (
+        <input type="button" value="Start Challenge" onClick={startChallenge} />
+      ) : (
+        <input type="button" value="Verify Code" onClick={verifyCode} />
+      )}
+    </>
+  )
+}
+```
+
+- You can extract the available MFA factors for the user by calling
+  `supabase.auth.mfa.listFactors()`. Don't worry this method is also very quick
+  and rarely uses the network.
+- If `listFactors()` returns more than one factor (or of a different type) you
+  should present the user with a choice. For simplicity this is not shown in
+  the example.
+- Phone numbers are unique per user. Users can only have one verified phone factor with a given phone number.
+  Attempting to enroll a new phone factor alongside an existing verified factor with the same number will result in an error.
+- Each time the user presses the "Submit" button a new challenge is created for
+  the chosen factor (in this case the first one)
+- On successful verification, the client library will refresh the session in
+  the background automatically and finally call the `onSuccess` callback, which
+  will show the authenticated `App` component on screen.
+
+### Security Configuration
+
+In Supabase Auth, each code has a default validity of `300s` and successive codes remain valid until expiry. For added security, we recommend increasing the OTP Length under `Settings > Authentication > Advanced MFA`.
```

Comment:
```suggestion
Each code is valid for up to 5 minutes, after which a new one can be sent. Successive codes remain valid until expiry. When possible choose the longest code length acceptable to your use case, at a minimum of 6. This can be configured in the [Authentication Settings](/dashboard/project/_/settings/auth).
```